diff -Naurp git_org/Makefile.inc git/Makefile.inc
--- git_org/Makefile.inc	2016-02-17 20:50:48.948804266 -0800
+++ git/Makefile.inc	2016-02-17 20:52:17.491262041 -0800
@@ -13,20 +13,6 @@ ifeq ($(TOPDIR),)
 	TOPDIR	= ..
 endif
 
-ifndef LIB
-	ifeq ($(shell test -d /lib64 && echo 1),1)
-		LIB=lib64
-	else
-		LIB=lib
-	endif
-endif
-
-ifndef SYSTEMD
-	ifeq ($(shell systemctl --version > /dev/null 2>&1 && echo 1), 1)
-		SYSTEMD = $(shell systemctl --version 2> /dev/null |  sed -n 's/systemd \([0-9]*\)/\1/p')
-	endif
-endif
-
 prefix      = 
 exec_prefix = $(prefix)
 bindir      = $(exec_prefix)/sbin
