diff -Naurp samhain-4.1.0_org/configure samhain-4.1.0/configure
--- samhain-4.1.0_org/configure	2016-02-29 02:52:31.274782918 -0500
+++ samhain-4.1.0/configure	2016-02-29 02:54:13.238846319 -0500
@@ -10469,13 +10469,21 @@ fi
 done
 
 	  else
-	    pg_lib_dir=`${PGCONF} --libdir`
+	    if test "x$cross_compiling" = "xyes"; then
+  	      pg_lib_dir="=${libdir}"
+	    else
+	      pg_lib_dir=`${PGCONF} --libdir`
+	    fi
 	    if test x"$enable_static" = xyes; then
 	      LIBS="$LIBS -L${pg_lib_dir} -lpq -lcrypt -lm"
             else
 	      LIBS="$LIBS -L${pg_lib_dir} -lpq -lm"
             fi
-	    pg_inc_dir=`${PGCONF} --includedir`
+	    if test "x$cross_compiling" = "xyes"; then
+	      pg_inc_dir="=${includedir}"
+	    else  
+	      pg_inc_dir=`${PGCONF} --includedir`
+	    fi
 	    # CFLAGS="$CFLAGS -I${pg_inc_dir}"
 	    CPPFLAGS="$CPPFLAGS -I${pg_inc_dir}"
 	  fi
