--- ACE_wrappers.org/include/makeinclude/platform_macros.GNU	2012-09-06 16:10:09.955785098 +0530
+++ ACE_wrappers/include/makeinclude/platform_macros.GNU	2012-09-20 21:03:09.379025295 +0530
@@ -0,0 +1,5 @@
+# Work-around #593225
+ARMEL_TARGET := $(shell echo '__ARMEL__' | $(CC) -E - | tail -n 1)
+ifeq ($(ARMEL_TARGET),1)
+  no_hidden_visibility = 1
+endif
diff -Naur ACE_wrappers_org/include/makeinclude/platform_linux_common.GNU ACE_wrappers/include/makeinclude/platform_linux_common.GNU
--- ACE_wrappers_org/include/makeinclude/platform_linux_common.GNU	2012-11-20 15:26:17.390087980 +0530
+++ ACE_wrappers/include/makeinclude/platform_linux_common.GNU	2012-11-20 15:27:25.682090639 +0530
@@ -85,7 +85,7 @@
 endif
 
 PLATFORM_AIO_SUPPORT := \
-       $(shell test "`ls -L /usr/lib*/librt.so* /lib*/librt.so*`" && echo -DACE_HAS_AIO_CALLS)
+       $(shell test "`ls -L $(STAGING_DIR_TARGET)/usr/lib*/librt.so* $(STAGING_DIR_TARGET)/lib*/librt.so*`" && echo -DACE_HAS_AIO_CALLS)
 
 GNU_LIBPTHREAD_VERSION := $(shell getconf GNU_LIBPTHREAD_VERSION 2> /dev/null || echo Unknown)
 ifeq (NPTL, $(word 1,$(GNU_LIBPTHREAD_VERSION)))
diff -Naur ACE_wrappers_org/include/makeinclude/platform_linux.GNU ACE_wrappers/include/makeinclude/platform_linux.GNU
--- ACE_wrappers_org/include/makeinclude/platform_linux.GNU	2012-11-20 15:26:17.390087980 +0530
+++ ACE_wrappers/include/makeinclude/platform_linux.GNU	2012-11-20 15:26:59.106089606 +0530
@@ -80,7 +80,7 @@
 ifeq ($(threads),1)
   LIBS += -lpthread
   # look for the rt library in the usual places
-  LIBS += $(shell test "`ls -L /usr/lib*/librt.so* /lib*/librt.so*`" && echo -lrt)
+  LIBS += $(shell test "`ls -L $(STAGING_DIR_TARGET)/usr/lib*/librt.so* $(STAGING_DIR_TARGET)/lib*/librt.so*`" && echo -lrt)
 endif
 
 ifeq ($(optimize),1)
