--- ACE_wrappers-orig/ace/os_include/os_sched.h	2009-04-17 14:04:36.000000000 +0000
+++ ACE_wrappers/ace/os_include/os_sched.h	2017-12-21 23:43:30.411595586 +0000
@@ -36,17 +36,6 @@
 {
 #endif /* __cplusplus */
 
-#if !defined (__cpu_set_t_defined) || !defined (ACE_HAS_CPU_SET_T)
-#if defined (ACE_HAS_CPUSET_T)
-   typedef cpuset_t cpu_set_t;
-#else
-#  define ACE_CPU_SETSIZE 1024
-   typedef struct
-   {
-     ACE_UINT32 bit_array_[ACE_CPU_SETSIZE / (8 * sizeof (ACE_UINT32))];
-   } cpu_set_t;
-#endif
-#endif /* !ACE_HAS_CPU_SET_T || !__cpu_set_t_defined */
 
 #ifdef __cplusplus
 }
