diff -Nuarp scsirastools-1.6.4_org/src/getmd.c scsirastools-1.6.4/src/getmd.c
--- scsirastools-1.6.4_org/src/getmd.c	2011-08-23 18:56:05.000000000 +0530
+++ scsirastools-1.6.4/src/getmd.c	2012-10-08 19:23:37.172354489 +0530
@@ -82,7 +82,7 @@ static int findmatch(const char *buffer,
     } else return (-1);  /*not found*/
 }
 
-static int get_line(FILE * fd, char *buf, int len)
+static int get_line_local(FILE * fd, char *buf, int len)
 {
     int i;
     int ch;
@@ -120,7 +120,7 @@ int getmd(char *devpattn, int sdevpattn,
     sraidstr = strlen(raidstr);
     strcpy(mdpart, "/dev/sda1");	/* default */
     strcpy(rdev, "/dev/md0");	/* default */
-    while ((rlen = get_line(fd, buf, 80)) > 0) {	/* process each line */
+    while ((rlen = get_line_local(fd, buf, 80)) > 0) {	/* process each line */
 	buf[rlen] = 0;		/* stringify */
 	/* look for raiddev string */
 	i = findmatch(buf, rlen, raidstr, sraidstr, 0);
diff -Nuarp scsirastools-1.6.4_org/src/sgraidmon.c scsirastools-1.6.4/src/sgraidmon.c
--- scsirastools-1.6.4_org/src/sgraidmon.c	2011-08-23 18:56:05.000000000 +0530
+++ scsirastools-1.6.4/src/sgraidmon.c	2012-10-08 19:24:42.116354803 +0530
@@ -330,7 +330,7 @@ getmd(char *devname, char *mddev, char *
     rdev[0] = 0;	/* default rdev (mddev) */
     strcpy(mdpart, "/dev/sdz1");	/* default mdpart */
     foundit = 0;
-    while ((rlen = get_line(fd, buf, 80)) > 0) {	/* process each line */
+    while ((rlen = get_line_local(fd, buf, 80)) > 0) {	/* process each line */
 	buf[rlen] = 0;		/* stringify */
 	/* check for comment */
 	if (buf[0] == '#') continue;
@@ -392,7 +392,7 @@ int getnpart(void)
    if (fd == NULL) return (np);
    sraidstr = strlen(raidstr);
    sdevstr = strlen(devstr);
-   while ((rlen = get_line(fd, buf, 80)) > 0) {	/* process each line */
+   while ((rlen = get_line_local(fd, buf, 80)) > 0) {	/* process each line */
 	buf[rlen] = 0;		/* stringify */
 	/* check for comment */
 	if (buf[0] == '#') continue;
diff -Nuarp scsirastools-1.6.4_org/src/sgsafte.c scsirastools-1.6.4/src/sgsafte.c
--- scsirastools-1.6.4_org/src/sgsafte.c	2011-08-23 18:56:05.000000000 +0530
+++ scsirastools-1.6.4/src/sgsafte.c	2012-10-08 19:25:09.860354939 +0530
@@ -231,7 +231,7 @@ getmd(char *devname, char *mddev, char *
     rdev[0] = 0;	/* default rdev (mddev) */
     strcpy(mdpart, "/dev/sdz1");	/* default mdpart */
     foundit = 0;
-    while ((rlen = get_line(fd, buf, 80)) > 0) {	/* process each line */
+    while ((rlen = get_line_local(fd, buf, 80)) > 0) {	/* process each line */
 	buf[rlen] = 0;		/* stringify */
 	/* look for raiddev string */
 	i = findmatch(buf, rlen, raidstr, sraidstr, 0);
@@ -284,7 +284,7 @@ int getnpart(void)
    fd = fopen(raidfile, "r");
    if (fd == NULL) return (np);
    sraidstr = strlen(raidstr);
-   while ((rlen = get_line(fd, buf, 80)) > 0) {	/* process each line */
+   while ((rlen = get_line_local(fd, buf, 80)) > 0) {	/* process each line */
 	buf[rlen] = 0;		/* stringify */
 	/* look for raiddev string */
 	i = findmatch(buf, rlen, raidstr, sraidstr, 0);
diff -Nuarp scsirastools-1.6.4_org/src/sgsubmon.c scsirastools-1.6.4/src/sgsubmon.c
--- scsirastools-1.6.4_org/src/sgsubmon.c	2011-08-23 18:56:05.000000000 +0530
+++ scsirastools-1.6.4/src/sgsubmon.c	2012-10-08 19:25:32.756355052 +0530
@@ -186,7 +186,7 @@ void dump_log(uchar * pbuf, int sz, char
     dump_buf(fd, pbuf,sz,tag,fshowascii);
 }
 
-int get_line(FILE * fd, char *buf, int len)
+int get_line_local(FILE * fd, char *buf, int len)
 {
     int i;
     int ch;
diff -Nuarp scsirastools-1.6.4_org/src/sgsubmon.h scsirastools-1.6.4/src/sgsubmon.h
--- scsirastools-1.6.4_org/src/sgsubmon.h	2011-08-23 18:56:05.000000000 +0530
+++ scsirastools-1.6.4/src/sgsubmon.h	2012-10-08 19:26:07.072355216 +0530
@@ -129,7 +129,7 @@ void itoh(uchar * chp, int len, char *st
 void dumpbufr(FILE * fdout, uchar * bufp, int mlen, char *hdr);
 void dump_buf(FILE * fdout, uchar * bufp, int mlen, char *hdr, char fascii);
 void dump_log(uchar * bufp, int mlen, char *hdr, char fascii);
-int  get_line(FILE * fd, char *buf, int len) ;
+int  get_line_local(FILE * fd, char *buf, int len) ;
 int  findmatch(char *buffer,int sbuf, char *pattern,int spattern,char figncase);
 int  fillstr(char *to, uchar *from,int len);
 int  IsSATA(int idev);
