diff -Naurp openl2tp-1.8_org/Makefile openl2tp-1.8/Makefile
--- openl2tp-1.8_org/Makefile	2015-07-24 10:30:22.116360438 +0530
+++ openl2tp-1.8/Makefile	2015-07-24 10:34:34.484800364 +0530
@@ -225,19 +225,19 @@ endif
 %_client.o:		%_client.c
 			$(CC) -I. -MMD -w $(CFLAGS.optimize) -c $(ADD_CPPFLAGS) $<
 
-%_server.o:		%_server.c
+%_server.o:		%_server.c %.h
 			$(CC) -I. -MMD -w $(CFLAGS.optimize) -c $(ADD_CPPFLAGS) $<
 
-%_xdr.c:		%.x
+%_xdr.c:		%.x %.h
 			-$(RM) $@
 			$(RPCGEN) $(RPCGENFLAGS) -c -o $@ $<
 
-%_server.c:		%.x
+%_server.c:		%.x %.h
 			-$(RM) $@ $@.tmp
 			$(RPCGEN) $(RPCGENFLAGS) -m -o $@.tmp $<
 			cat $@.tmp | sed -e 's/switch (rqstp->rq_proc) {/if (l2tp_api_rpc_check_request(transp) < 0) return; switch (rqstp->rq_proc) {/' > $@
 
-%_client.c:		%.x
+%_client.c:		%.x %.h
 			-$(RM) $@
 			$(RPCGEN) $(RPCGENFLAGS) -l -o $@ $<
 
@@ -269,9 +269,10 @@ openl2tpd:		$(L2TPD_SRCS.o)
 			$(CC) -o $@ $^ $(LDFLAGS.l2tpd)
 
 l2tpconfig:		$(L2TPCONFIG_SRCS.o)
+			$(MAKE) -C cli $(MFLAGS) libcli.a
 			$(CC) -o $@ $^ $(LDFLAGS.l2tpconfig)
 
-%.o:	%.c
+%.o:	%.c l2tp_private.h
 			$(CC) -c $(CFLAGS) $(ADD_CFLAGS) $< -o $@
 
 l2tp_options.h:	FORCE
@@ -318,6 +319,8 @@ endif
 		mv $@.tmp $@ ;\
 	fi
 
+l2tp_private.h: l2tp_options.h l2tp_avp.h l2tp_rpc.h
+
 .PHONY:			all install-all install-daemon install-app
 
 install:		install-all
