diff -Naurp policycoreutils-2.4_org/hll/pp/pp.c policycoreutils-2.4/hll/pp/pp.c
--- policycoreutils-2.4_org/hll/pp/pp.c	2015-10-21 02:40:27.872123445 -0400
+++ policycoreutils-2.4/hll/pp/pp.c	2015-10-21 02:59:44.647776184 -0400
@@ -32,6 +32,7 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <unistd.h>
+#include <features.h>
 
 #include <sepol/module.h>
 #include <sepol/policydb/conditional.h>
@@ -2810,7 +2811,12 @@ static int level_string_to_cil(char *lev
 	char *token = NULL;
 	char *ranged = NULL;
 
+#if __GLIBC_PREREQ (2,7)
 	matched = sscanf(levelstr, "%m[^:]:%ms", &sens, &cats);
+#else
+	matched = sscanf(levelstr, "%a[^:]:%as", &sens, &cats);
+#endif /* __GLIBC_PREREQ (2,7) */
+
 	if (matched < 1 || matched > 2) {
 		log_err("Invalid level: %s", levelstr);
 		rc = -1;
@@ -2875,7 +2881,12 @@ static int context_string_to_cil(char *c
 	char *type = NULL;
 	char *level = NULL;
 
+#if __GLIBC_PREREQ (2,7)
 	matched = sscanf(contextstr, "%m[^:]:%m[^:]:%m[^:]:%ms", &user, &role, &type, &level);
+#else
+	matched = sscanf(contextstr, "%a[^:]:%a[^:]:%a[^:]:%as", &user, &role, &type, &level);
+#endif /* __GLIBC_PREREQ (2,7) */
+
 	if (matched < 3 || matched > 4) {
 		log_err("Invalid context: %s", contextstr);
 		rc = -1;
@@ -2937,7 +2948,11 @@ static int seusers_to_cil(struct sepol_m
 			continue;
 		}
 
+#if __GLIBC_PREREQ (2,7)
 		matched = sscanf(buf, "%m[^:]:%m[^:]:%ms", &user, &seuser, &level);
+#else
+		matched = sscanf(buf, "%a[^:]:%a[^:]:%as", &user, &seuser, &level);
+#endif /* #if __GLIBC_PREREQ (2,7) */
 
 		if (matched < 2 || matched > 3) {
 			log_err("Invalid seuser line: %s", line);
@@ -3020,7 +3035,12 @@ static int user_extra_to_cil(struct sepo
 	while ((line_len = getline(&line, &len, fp)) != -1) {
 		line[line_len - 1] = '\0';
 
+#if __GLIBC_PREREQ (2,7)
 		matched = sscanf(line, "user %ms prefix %m[^;];", &user, &prefix);
+#else
+		matched = sscanf(line, "user %as prefix %a[^;];", &user, &prefix);
+#endif /* __GLIBC_PREREQ (2,7) */
+
 		if (matched != 2) {
 			rc = -1;
 			log_err("Invalid file context line: %s", line);
@@ -3084,7 +3104,12 @@ static int file_contexts_to_cil(struct s
 			continue;
 		}
 
+#if __GLIBC_PREREQ (2,7)
 		matched = sscanf(buf, "%ms %ms %ms", &regex, &mode, &context);
+#else
+		matched = sscanf(buf, "%as %as %as", &regex, &mode, &context);
+#endif /* __GLIBC_PREREQ (2,7) */
+
 		if (matched < 2 || matched > 3) {
 			rc = -1;
 			log_err("Invalid file context line: %s", line);
