diff -ur .orig-hpitest-3.1.1/src/framework/LocationMgr.cc hpitest-3.1.1/src/framework/LocationMgr.cc
--- .orig-hpitest-3.1.1/src/framework/LocationMgr.cc	2010-04-27 12:05:23.000000000 +0400
+++ hpitest-3.1.1/src/framework/LocationMgr.cc	2011-03-20 12:05:10.406767998 +0300
@@ -17,6 +17,8 @@
  * Author(s):
  *     Donald A. Barre <dbarre@unh.edu>
  */
+
+#include <cstdio>
 #include <cstring>
 #include <string.h>
 
diff -ur .orig-hpitest-3.1.1/src/framework/UserPrompter.cc hpitest-3.1.1/src/framework/UserPrompter.cc
--- .orig-hpitest-3.1.1/src/framework/UserPrompter.cc	2010-04-27 12:05:23.000000000 +0400
+++ hpitest-3.1.1/src/framework/UserPrompter.cc	2011-03-20 12:05:32.887768003 +0300
@@ -20,6 +20,7 @@
 
 #include <stdlib.h>
 
+#include <cstdio>
 #include <cstring>
 #include <iostream>
 #include <limits>
diff -ur .orig-hpitest-3.1.1/src/hpi/HpiString.cc hpitest-3.1.1/src/hpi/HpiString.cc
--- .orig-hpitest-3.1.1/src/hpi/HpiString.cc	2010-04-27 12:05:23.000000000 +0400
+++ hpitest-3.1.1/src/hpi/HpiString.cc	2011-03-20 11:59:31.801768003 +0300
@@ -24,6 +24,7 @@
  * 	            enhancements to be HPI-B.03.01 compliant
  */
 
+#include <cstdio>
 #include <cstring>
 
 #include "HpiString.h"
diff -ur .orig-hpitest-3.1.1/src/tests/version/saHpiInitialize/InvalidData.cc hpitest-3.1.1/src/tests/version/saHpiInitialize/InvalidData.cc
--- .orig-hpitest-3.1.1/src/tests/version/saHpiInitialize/InvalidData.cc	2010-04-27 12:05:22.000000000 +0400
+++ hpitest-3.1.1/src/tests/version/saHpiInitialize/InvalidData.cc	2011-03-20 12:00:40.123768000 +0300
@@ -18,6 +18,8 @@
  *     Lars Wetzel <larswetzel@users.sourceforge.net>
  */
 
+#include <cstdlib>
+
 #include "InvalidData.h"
 
 using namespace ns_saHpiInitialize;
