--- a/Makefile.am.orig	2012-11-17 10:34:59.565947622 +0530
+++ b/Makefile.am	2012-11-17 10:35:34.967947622 +0530
@@ -226,9 +226,9 @@
 		sed -i 's|xxCLCCLIDIRxx|$(pkgclcclidir)|g' "$$i"; \
 	done
 	@for i in $$(grep -lr -e 'xxBINDIRxx' -e 'xxLIBDIRxx' -e 'xxCLCCLIDIRxx' $(DESTDIR)$(pkgsysconfdir)/*) ; do \
-		sed --follow-symlinks -i 's|xxBINDIRxx|$(bindir)|g' "$$i"; \
-		sed --follow-symlinks -i 's|xxLIBDIRxx|$(pkglibdir)|g' "$$i"; \
-		sed --follow-symlinks -i 's|xxCLCCLIDIRxx|$(pkgclcclidir)|g' "$$i"; \
+		sed -i 's|xxBINDIRxx|$(bindir)|g' "$$(readlink -f $$i)"; \
+		sed -i 's|xxLIBDIRxx|$(pkglibdir)|g' "$$(readlink -f $$i)"; \
+		sed -i 's|xxCLCCLIDIRxx|$(pkgclcclidir)|g' "$$(readlink -f $$i)"; \
 	done
 
 uninstall-hook:
