--- opendiameter-1.0.7-i_org/configure.in	2015-02-25 17:22:25.357894707 +0530
+++ opendiameter-1.0.7-i/configure.in	2015-02-25 17:23:17.213946190 +0530
@@ -60,6 +60,7 @@ AM_CONDITIONAL(compile_EAPTLS, [test "$w
 # Checks for programs.
 AC_PROG_CC
 AC_PROG_CXX
+AC_SUBST(_LT_TAGVAR(postdeps, CXX),["-lcrypto -lACEXML_Parser -lboost_system $postdeps_CXX"])
 AC_PROG_INSTALL
 AC_PROG_LIBTOOL
 AM_PROG_CC_C_O
