diff -upNr opendiameter-1.0.7-i.orig/libdiameter/include/aaa_transport_ace.h opendiameter-1.0.7-i/libdiameter/include/aaa_transport_ace.h
--- opendiameter-1.0.7-i.orig/libdiameter/include/aaa_transport_ace.h	2011-02-18 05:58:32.000000000 -0500
+++ opendiameter-1.0.7-i/libdiameter/include/aaa_transport_ace.h	2011-02-18 05:59:49.000000000 -0500
@@ -39,6 +39,7 @@
 #include "ace/SOCK_SEQPACK_Acceptor.h"
 #include "ace/SOCK_SEQPACK_Connector.h"
 #include "ace/Signal.h"
+#include "ace/Sig_Handler.h"
 #include "ace/Handle_Set.h"
 #include "aaa_transport_interface.h"
 
diff -upNr opendiameter-1.0.7-i.orig/libpana/include/pana_serial_num.h opendiameter-1.0.7-i/libpana/include/pana_serial_num.h
--- opendiameter-1.0.7-i.orig/libpana/include/pana_serial_num.h	2011-02-18 05:58:32.000000000 -0500
+++ opendiameter-1.0.7-i/libpana/include/pana_serial_num.h	2011-02-18 06:00:45.000000000 -0500
@@ -205,7 +205,8 @@ class PANA_SerialNumber
         }
         static inline ACE_UINT32 GenerateISN(ACE_UINT32 seed = 0)  {
             if (seed == 0) {
-                ACE_System_Time::get_local_system_time(seed);
+		time_t t = 0;
+                ACE_System_Time::get_local_system_time(t);
             }
             // simple time seeded randon number generator
             ACE_OS::srand(seed + ACE_OS::rand());
