--- opendiameter-1.0.7-i/libdiametermip4/Makefile.am_org	2012-11-21 10:10:49.976715253 +0530
+++ opendiameter-1.0.7-i/libdiametermip4/Makefile.am	2012-11-21 10:11:01.416715698 +0530
@@ -41,7 +41,6 @@
 		   include/diameter_mip4_aaas_server_interface.hxx \
 		   include/diameter_mip4_aaas_server_sint_interface.hxx \
 		   include/mip4_diameter_fa_client_interface.hxx \
-		   include/mip4_diameter_ha_client_interface.hxx \
                    include/diameter_mip4_parser.hxx 
 
 lib_LTLIBRARIES = libdiametermip4.la
diff -Naur opendiameter-1.0.7-i_org/Makefile.am opendiameter-1.0.7-i/Makefile.am
--- opendiameter-1.0.7-i_org/Makefile.am	2012-11-21 15:27:31.273455132 +0530
+++ opendiameter-1.0.7-i/Makefile.am	2012-11-21 15:29:00.213458595 +0530
@@ -3,14 +3,6 @@
        $(top_srcdir)/docs/development_policy.txt \
        $(top_srcdir)/docs/coding_style.txt \
        $(top_srcdir)/docs/bugtracking-diagram.pdf \
-       $(top_srcdir)/docs/framework \
-       $(top_srcdir)/docs/libdiameter \
-       $(top_srcdir)/docs/libdiametereap \
-       $(top_srcdir)/docs/libdiameternasreq \
-       $(top_srcdir)/docs/libdiametermip4 \
-       $(top_srcdir)/docs/libeap \
-       $(top_srcdir)/docs/libpana \
-       $(top_srcdir)/docs/nasreq \
        $(top_srcdir)/opendiameter.kdevelop \
        $(top_srcdir)/tools/getopt.patch \
        $(top_srcdir)/tools/getopt1.patch \
@@ -18,8 +10,16 @@
        $(top_srcdir)/RELEASE \
        $(top_srcdir)/README
 
+MISC_EXTRA = $(top_srcdir)/docs/framework \
+       $(top_srcdir)/docs/libdiameter \
+       $(top_srcdir)/docs/libdiametereap \
+       $(top_srcdir)/docs/libdiameternasreq \
+       $(top_srcdir)/docs/libdiametermip4 \
+       $(top_srcdir)/docs/libeap \
+       $(top_srcdir)/docs/libpana \
+       $(top_srcdir)/docs/nasreq 
 
-EXTRA_DIST = $(MISC)
+EXTRA_DIST = $(MISC) $(MISC_EXTRA)
 
 datadir = $(prefix)/share/opendiameter/doc
 data_DATA = $(MISC)
diff -Naur opendiameter-1.0.7-i_org/Makefile.in opendiameter-1.0.7-i/Makefile.in
--- opendiameter-1.0.7-i_org/Makefile.in	2012-11-21 15:27:31.289455133 +0530
+++ opendiameter-1.0.7-i/Makefile.in	2012-11-21 15:30:12.645461416 +0530
@@ -215,7 +215,16 @@
        $(top_srcdir)/RELEASE \
        $(top_srcdir)/README
 
-EXTRA_DIST = $(MISC)
+MISC_EXTRA = $(top_srcdir)/docs/framework \
+       $(top_srcdir)/docs/libdiameter \
+       $(top_srcdir)/docs/libdiametereap \
+       $(top_srcdir)/docs/libdiameternasreq \
+       $(top_srcdir)/docs/libdiametermip4 \
+       $(top_srcdir)/docs/libeap \
+       $(top_srcdir)/docs/libpana \
+       $(top_srcdir)/docs/nasreq \
+
+EXTRA_DIST = $(MISC) $(MISC_EXTRA)
 data_DATA = $(MISC)
 SUBDIRS_LIBS = libodutl \
                libdiamparser \
--- opendiameter-1.0.7-i/Makefile.am_org	2012-11-21 16:17:05.761570954 +0530
+++ opendiameter-1.0.7-i/Makefile.am	2012-11-21 16:17:23.841571658 +0530
@@ -49,4 +49,5 @@
 dist-hook:
 	rm -rf `find $(distdir) -name CVS`
 
+all-am: Makefile $(DATA) $(HEADERS) $(MISC_EXTRA)
 
