--- ipmiutil-2.9.6/util/Makefile.am_orig	2015-04-22 23:48:27.000000000 +0530
+++ ipmiutil-2.9.6/util/Makefile.am	2015-07-28 22:16:19.832645611 +0530
@@ -51,7 +51,7 @@ LDADD += @LANDESK_LDADD@
 # Could also use CMDMOD += ipmilan2.c  instead, if completed.
 AM_CPPFLAGS += @LANPLUS_CFLAGS@ 
 LDADD += @LANPLUS_LIB@ @LANPLUS_CRYPTO@ 
-LANPLUS_OBJ = $(shell ar t @LANPLUS_LIB@ 2>/dev/null) 
+LANPLUS_OBJ = $(shell $(AR) t @LANPLUS_LIB@ 2>/dev/null) 
 
 # sbin_PROGRAMS are built by default and copied to /usr/sbin at install time
 # EXTRA_PROGRAMS are built by default but not included in the install package
@@ -82,7 +82,7 @@ $(DEV_LIB):	$(CMDMOD:.c=.o) @LANPLUS_LIB
 	if [ "x@LANPLUS_SAM@" = "xyes" ]; then \
 	   $(CC) $(CFLAGS_SAMX) -o $(tmpobj)/ipmilanplus.o -c ipmilanplus.c; \
 	   $(CC) $(CFLAGS_SAMX) -o $(tmpobj)/ipmilan.o -c ipmilan.c; \
-	   ar x @LANPLUS_LIB@ ; \
+	   $(AR) x @LANPLUS_LIB@ ; \
 	   $(AR) cru $(DEV_LIB) $(LIBOBJ) $(LANPLUS_OBJ); \
 	   $(RANLIB) $(DEV_LIB); \
 	else   \
@@ -97,7 +97,7 @@ $(SHR_LIB):	$(CMDMOD:.c=.o) @LANPLUS_LIB
 	if [ "x@LANPLUS_SAM@" = "xyes" ]; then \
 	   $(CC) $(CFLAGS_SAMX) -o $(tmpobj)/ipmilanplus.o -c ipmilanplus.c; \
 	   $(CC) $(CFLAGS_SAMX) -o $(tmpobj)/ipmilan.o -c ipmilan.c; \
-	   ar x @LANPLUS_LIB@ ; \
+	   $(AR) x @LANPLUS_LIB@ ; \
 	   $(CC) $(LDFLAGS) -shared -o $(SHR_LIB) $(LIBOBJ) $(LANPLUS_OBJ); \
 	else \
 	   $(CC) $(CFLAGS_SAM) -o $(tmpobj)/ipmilanplus.o -c ipmilanplus.c; \
