diff -ur openhpi-2.14.1/acinclude.m4 openhpi-2.14.1/acinclude.m4
--- openhpi-2.14.1/acinclude.m4	2009-11-23 23:19:38.000000000 +0300
+++ openhpi-2.14.1/acinclude.m4	2010-09-22 12:46:36.203497121 +0400
@@ -162,8 +162,8 @@
     ],
     [
         have_netsnmp=yes
-        SNMPFLAGS=`net-snmp-config --cflags | perl -p -e 's/-O\S*//g'`
-        SNMPLIBS=`net-snmp-config --libs`
+        SNMPFLAGS=`${net_snmp_config:-net-snmp-config} --cflags | perl -p -e 's/-O\S*//g'`
+        SNMPLIBS=`${net_snmp_config:-net-snmp-config} --libs`
         AC_MSG_RESULT(yes)
     ],
     [AC_MSG_RESULT(no.  No SNMP based plugins can be built!)])
diff -ur openhpi-2.14.1/configure.ac openhpi-2.14.1/configure.ac
--- openhpi-2.14.1/configure.ac	2009-11-23 23:19:38.000000000 +0300
+++ openhpi-2.14.1/configure.ac	2010-09-22 13:44:02.105137327 +0400
@@ -148,6 +148,10 @@
 AC_CHECK_LIB([sysfs], [sysfs_get_mnt_path], [have_sysfs=yes])
 AC_CHECK_HEADER([sysfs/libsysfs.h], [], [have_sysfs=no])
 
+AC_ARG_WITH(net-snmp-config,
+    [[  --with-net-snmp-config=SCRIPT   run SCRIPT as net-snmp-config]],
+    [net_snmp_config=$withval])
+
 OH_CHECK_NETSNMP
 
 AC_CHECK_LIB([uuid], [uuid_generate], [have_uuid=yes], [
@@ -164,7 +168,7 @@
 
 dnl xml is used for XML-based communication in ilo2_ribcl and oa_soap
 AC_CHECK_LIB([xml2],[xmlParseMemory],[XML2_LIB=-lxml2],[XML2_LIB=])
-AC_CHECK_HEADERS([libxml2/libxml/xmlexports.h],[XML2_INCLUDE="-I/usr/include/libxml2"],[XML2_INCLUDE])
+AC_CHECK_HEADERS([libxml2/libxml/xmlexports.h],[XML2_INCLUDE="-I${PKG_CONFIG_SYSROOT_DIR}/usr/include/libxml2"],[XML2_INCLUDE])
 AC_SUBST(XML2_LIB)
 AC_SUBST(XML2_INCLUDE)
 
