--- openhpi-2.14.1/plugins/ipmidirect/ipmi_mc_vendor.cpp	2011-02-22 13:19:16.023983002 +0300
+++ openhpi-2.14.1/plugins/ipmidirect/ipmi_mc_vendor.cpp	2011-02-22 13:19:25.365983049 +0300
@@ -931,7 +931,7 @@
           /* Everything is valid, create the Watchdog RDR */
           stdlog << "CreateWatchdogs Resource type " << res->EntityPath().GetEntryType(0) << " instance " << res->EntityPath().GetEntryInstance(0) << "\n";
         
-          cIpmiRdr *wd = new cIpmiWatchdog::cIpmiWatchdog( mc, SAHPI_DEFAULT_WATCHDOG_NUM, 0 );
+          cIpmiRdr *wd = new cIpmiWatchdog( mc, SAHPI_DEFAULT_WATCHDOG_NUM, 0 );
         
           wd->EntityPath() = res->EntityPath();
         
